tests/core/test_state_sync.py

```diff
@@ -1341,5 +1360,55 @@ def test_get_snapshots(mocker):
     cache.get_snapshots([])
     mock.get_snapshots.assert_not_called()
 
-    cache.get_snapshots(None)
-    mock.get_snapshots.assert_called()
+
+def test_snapshot_batching(state_sync, mocker, make_snapshot):
+    mock = mocker.Mock()
+
+    state_sync.SNAPSHOT_BATCH_SIZE = 2
+    state_sync.engine_adapter = mock
+
+    state_sync.delete_snapshots(
+        (
+            SnapshotId(name="a", identifier="1"),
+            SnapshotId(name="a", identifier="2"),
+            SnapshotId(name="a", identifier="3"),
+        )
+    )
+    calls = mock.delete_from.call_args_list
+    assert len(calls) == 2
+    assert calls[0][1] == {"where": parse_one("(name, identifier) in (('a', '1'), ('a', '2'))")}
+    assert calls[1][1] == {"where": parse_one("(name, identifier) in (('a', '3'))")}
+
+    mock.fetchall.side_effect = [
+        [
+            [
+                make_snapshot(
+                    SqlModel(name="a", query=parse_one("select 1")),
+                ).json(),
+            ],
+            [
+                make_snapshot(
+                    SqlModel(name="a", query=parse_one("select 2")),
+                ).json(),
+            ],
+        ],
+        [
+            [
+                make_snapshot(
+                    SqlModel(name="a", query=parse_one("select 3")),
+                ).json(),
+            ],
+        ],
+    ]
+
+    snapshots = state_sync._get_snapshots(
```

Comment:
[Nit] Maybe a test for intervals too?

Reply:
i added modified the an interval test to make sure that code path is hit